app/services/bos/client.rb

```diff
@@ -0,0 +1,92 @@
+# frozen_string_literal: true
+
+require "faraday"
+require "json"
+
+class Bos::Client
+  API_PREFIX = "/api"
+
+  def initialize
+    @bos_enabled = Rails.configuration.bos_enabled
+
+    @conn =
+      Faraday.new(url: Rails.configuration.bos_base_url) do |f|
+        f.request :json
+        f.response :raise_error
+        f.response :json, content_type: /\bjson$/
+        f.headers["x-key"] = Rails.configuration.bos_api_key
+        f.headers["Content-Type"] = "application/json"
+        f.headers["Accept"] = "application/json"
+      end
+  end
+
+  def create_order(project_item)
+    post(
+      "#{API_PREFIX}/orders",
+      {
+        external_ref: project_item.iid.to_s,
+        project_ref: project_item.project_id.to_s,
+        config: {
+          params: serialize_order_params(project_item)
+        },
+        platforms: project_item.offer.service.platforms.map(&:name),
+        resource_ref: project_item.offer.service.friendly_id,
+        resource_type: project_item.offer.order_type,
+        resource_name: project_item.offer.service.name,
+        provider_pids:
+          project_item.offer.service.providers.map(&:pid).presence ||
+            [project_item.offer.service.resource_organisation.pid]
+      }
+    )
+  end
+
+  def create_message(message)
+    post(
+      "#{API_PREFIX}/messages",
+      {
+        content: message.message,
+        scope: "public",
```

Comment:
We have to think about proper messages scoping, but I think it's more business decision.